shared/middleware/channels_api.rb

```diff
@@ -59,6 +59,9 @@ class ChannelsApi < Sinatra::Base
 
     if src_channel
       data = storage_app.get(src_channel)
+      if data['name'].nil? and data['level'] == '/projects/algebra_game'
```

Comment:
This is actually a dead PR, and we ended up coming up with a better change that sets the name in the client.